module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -458,45 +471,56 @@ public function getHolding($bibId, array $patron = null, array $options = [])
      */
     public function patronLogin($username, $password)
     {
-        // Get user id
-        $query = ['query' => 'username == ' . $username];
-        $response = $this->makeRequest('GET', '/users', $query);
+        if ($this->config['User']['okapi_login'] ?? false) {
+            $tenant = $this->config['API']['tenant'];
+            $credentials = compact('tenant', 'username', 'password');
+            // Get token
+            try {
+                $response = $this->makeRequest(
+                    'POST',
+                    '/authn/login',
+                    json_encode($credentials)
+                );
+                $debugMsg = 'User logged in. User: ' . $username . '.';
+                // We've authenticated the user with Okapi, but we only have their
+                // username; set up a query to retrieve full info below.
+                $query = 'username == ' . $username;
+                // Replace admin with user as tenant if configured to do so:
+                if ($this->config['User']['use_user_token'] ?? false) {
+                    $this->token = $response->getHeaders()->get('X-Okapi-Token')
+                        ->getFieldValue();
+                    $debugMsg .= ' Token: ' . substr($this->token, 0, 30) . '...';
+                }
+                $this->debug($debugMsg);
+            } catch (Exception $e) {
+                return null;
+            }
+        } else {
+            // Construct user query using barcode, username, etc.
+            $usernameField = $this->config['User']['username_field'] ?? 'username';
```

Comment:
I moved these configuration loads down here since they only apply in this else clause... I also added comments to the .ini file noting that these only apply when okapi_login is false.